xfields/ibs/_equilibrium.py

```diff
@@ -253,59 +253,78 @@ def compute_emittance_evolution(
+    # fmt: off
+    # ---------------------------------------------------------------------------------------------
+    # Check for SR equilibrium emittances, damping constants and partition numbers in the TwissTable
+    _required_attrs = ["damping_constants_s", "partition_numbers", "eq_gemitt_x", "eq_gemitt_y", "eq_gemitt_zeta"]
+    if any(getattr(twiss, attr, None) is None for attr in _required_attrs):
+        LOGGER.error("Invalid TwissTable, does not have SR equilibrium properties. Did you configure radiation?")
+        raise AttributeError(
+            "The TwissTable must contain SR equilibrium emittances and damping constants. "
+            "Did you activate radiation and twiss with `eneloss_and_damping=True?`"
+        )
+    # # ---------------------------------------------------------------------------------------------
+    # # Perform checks on required & exclusive parameters
+    # assert total_beam_intensity is not None, "Must provide 'total_beam_intensity'"
+    # assert any([gemitt_x, nemitt_x]), "Must provide either 'gemitt_x' or 'nemitt_x'"
+    # assert any([gemitt_y, nemitt_y]), "Must provide either 'gemitt_y' or 'nemitt_y'"
+    # assert any([gemitt_zeta, nemitt_zeta]), "Must provide either 'gemitt_zeta' or 'nemitt_zeta'"
+    # if gemitt_x is not None:
+    #     assert nemitt_x is None, "Cannot provide both 'gemitt_x' and 'nemitt_x'"
+    # if gemitt_y is not None:
+    #     assert nemitt_y is None, "Cannot provide both 'gemitt_y' and 'nemitt_y'"
+    # if gemitt_zeta is not None:
+    #     assert nemitt_zeta is None, "Cannot provide both 'gemitt_zeta' and 'nemitt_zeta'"
+    # # ---------------------------------------------------------------------------------------------
+    # # Convert emittances to geometric if needed
+    # if nemitt_x is not None:
+    #     assert gemitt_x is None, "Cannot provide both 'gemitt_x' and 'nemitt_x'"
+    #     gemitt_x = nemitt_x / (twiss.beta0 * twiss.gamma0)
+    # if nemitt_y is not None:
+    #     assert gemitt_y is None, "Cannot provide both 'gemitt_y' and 'nemitt_y'"
+    #     gemitt_y = nemitt_y / (twiss.beta0 * twiss.gamma0)
+    # if nemitt_zeta is not None:
+    #     assert gemitt_zeta is None, "Cannot provide both 'gemitt_zeta' and 'nemitt_zeta'"
+    #     gemitt_zeta = nemitt_zeta / (twiss.beta0 * twiss.gamma0)
+    # ---------------------------------------------------------------------------------------------
+    # Check for valid value of emittance_constraint and warn if constraint provided but factor is 0
+    if emittance_constraint is not None:
+        assert emittance_constraint.lower() in ("coupling", "excitation"), "Invalid 'emittance_constraint', accepted values are 'coupling' or 'excitation'."
+        if emittance_coupling_factor == 0:
+            LOGGER.warning("As 'emittance_coupling_factor` is zero, providing 'emittance_constraint' has no effect!")
```

Comment:
Check the emittance constraint (if provided) is an accepted value. If so we also check if the factor is 0, in which case we go along but I log a warning to the user.